src/Profile.php

```diff
@@ -778,6 +778,363 @@ public function showForm($ID, array $options = [])
         return true;
     }
 
+    /**
+     * Get all rights to display for a specific form and interface.
+     *
+     * This is only used for GLPI core rights and not rights added by plugins.
+     *
+     * @param string $form The tab/form name
+     * @phpstan-param non-empty-string $form
+     * @param string $interface The interface name
+     * @phpstan-param 'central'|'helpdesk' $interface
+     * @return array
+     * @phpstan-return array<string, array{
+     *     rights?: array{}, itemtype?: class-string<CommonDBTM>, label: string, field: string, group: string, scope: string
+     * }>
+     * @internal BC not guaranteed. Only public so it can be used in tests to ensure search options are made for all rights.
+     */
+    public static function getRightsForForm(string $form, string $interface): array
+    {
+        // Helper function to streamline rights definition
+        /**
+         * @param class-string<CommonDBTM> $itemtype
+         * @param string|array $interface
+         * @param string $field
+         * @param array $options
+         * @return array
+         */
+        $fn_get_rights = static function (string $itemtype, $interface, string $field = null, array $options = []) {
+            $options = array_replace([
+                'group' => null,
+                'scope' => 'entity'
+            ], $options);
+            $rights = Profile::getRightsFor($itemtype, $interface);
+            if ($field === null) {
+                $field = $itemtype::$rightname;
+            }
+            return [
+                'rights' => $rights,
+                'label'  => $itemtype::getTypeName(Session::getPluralNumber()),
+                'field'  => $field,
+                'interface' => !is_array($interface) ? [$interface] : $interface,
+                'group' => $options['group'],
+                'scope' => $options['scope']
+            ];
+        };
+
+        static $all_rights = null;
+
+        if ($all_rights === null) {
+            $all_rights['tracking'] = [
```

Comment:
IMHO, declaration could be easier to read if it was a 3 levels array, like this:
```php
$all_rights = [
    'central' => [
        'tracking' => [
            'tickets' => [
                $fn_get_rights(Ticket::class),
                $fn_get_rights(TicketCost::class),
                $fn_get_rights(TicketRecurrent::class),
            ],
            'followups_tasks' => [...],
            ...
        ]
    ]
]
```

Reply:
I agree with that.
Finding where adding a new right will be easier so

Reply:
method getRightsForForm should permits also to filter by the 2 first level, and interface filtering should be in a third level